guides/micronaut-cloud-secrets-azure/micronaut-cloud-secrets-azure.adoc

```diff
@@ -0,0 +1,194 @@
+
+Create Azure Key Vault service resource.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-vault-resouce.png[]
+{empty} +
+
+When Key Vault service deployment is ready, go to the resource in order to create secrets. Prior to clicking on the Secrets option in the left menu list, copy the Vault URI from the main screen section. You will need to set it as an environment variable in order to integrate Azure Key Vault into your application.
+{empty} +
+
+image::azure-key-vault/azure-vault-obtain-vault-uri.png[]
+{empty} +
+
+Create new secrets for JDBC user and password. Click on the Secrets option in the left menu, then click on Generate/Import.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-secret-jdbc-user.png[]
+{empty} +
+
+image::azure-key-vault/azure-vault-create-secret-jdbc-password.png[]
+{empty} +
+
+Go to App registration in order to register your application with the Microsoft identity platform. After registering your application,
+copy the secret Application (client) ID, and Directory (tenant) ID. You will use them later on as environment variables in order to connect to Azure Key Vault.
+{empty} +
+
+image::azure-key-vault/azure-register-application.png[]
+{empty} +
+
+Navigate to your application profile in order to generate application credentials that you will use to authenticate with the Azure Key Vault service. In the left menu, click on the Certificates & secrets. To generate the application credentials, click on the Client secrets tab and then on the New client secret button. Copy the secret value. You will need to set it along the Application (client) ID, Directory (tenant) ID, and Vault URI as environment variables.
+NOTE: The client's secret value cannot be viewed, except immediately after creation. Be sure to save the secret when created before leaving the page
+{empty} +
+
+image::azure-key-vault/azure-register-application-add-client-secret.png[]
+{empty} +
+
+In order to finish the Key Vault setup process. you will need to assign a proper policy to the Key Vault resource. Go to your Key Vault resource profile. Click on the Access policies option in the left menu. Select Get and List options from the Secret Management Operations list. Finally, create an access policy.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-policy.png[]
+{empty} +
+
+Prior to completing the policy creation process, a security principal should be assigned. The policy should refer to your application acting as a security principal.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-principal.png[]
+{empty} +
+
+== Running the Application
+
+With almost everything in place, you can start the application and try it out. First, set environment variables to configure the application datasource, then start the application.
+
+Create environment variables for `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`,  `AZURE_SECRET_VALUE`, and `AZURE_VAULT_URI` which will be used in the Micronaut app's `application.yml` datasource:
```

Comment:
```suggestion
Create environment variables for `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`,  `AZURE_SECRET_VALUE`, and `AZURE_VAULT_URI`, which will be used in the Micronaut app's `application.yml` datasource:
```

Reply:
Covered in https://github.com/micronaut-projects/micronaut-guides/pull/1099/commits/27ffa927a580938d2c8edd451d078f4c25369a2f.